### ISSUE TITLE
[java] ConstructorCallsOverridableMethod occurs when unused overloaded method is defined

### ISSUE THREAD

**Issue by mistercam** [pmd/pmd#2348]
**Affects PMD Version:** 6.11.0, 6.21.0

**Rule:** [ConstructorCallsOverridableMethod](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#constructorcallsoverridablemethod)

**Description:**
If a constructor calls a non-overridable public method, and this method has an overridable overloaded version (which is not called from the constructor or from the non-overridable version,) then a ConstructorCallsOverridableMethod rule exception is raised.

**Code Sample demonstrating the issue:**

```java
public class Scratch {
    private Long member = 0L;

    public Scratch(final Long v) {
        setMember(v);
    }

    public final void setMember(final Long v) {
        this.member = v;
    }

    // Renaming this or marking it as final prevents the
    // 'ConstructorCallsOverridableMethod' rule from being broken.
    public void setMember(final String n) {
        setMember(Long.valueOf(n));
    }
}
```

**Running PMD through:** Gradle (with version 6.11.0) and CLI (with version 6.21.0)

**Comment by adangel** [comment]
Another example:

```java
import java.util.Date;

public class MyTimestamp {
    private final long t;
    public MyTimestamp() {
        t = new Date().getTime();
    }
    public long getTime() {
        return t;
    }
}
```
